experimental/C/src/AnytimePrime.lf

```diff
@@ -0,0 +1,67 @@
+/**
+ * This program is a concept demonstration showing how the check_deadline()
+ * function works. This program performs "anytime computation" (https://en.wikipedia.org/wiki/Anytime_algorithm)
+ * to find the largest prime within the given deadline. The check_deadline()
+ * function is called after checking each number if it is prime. If the
+ * check_deadline() is called after the deadline has passed, the function calls
+ * the deadline handler and returns true. Then the program outputs the largest
+ * prime found by that time and exits.
+ *
+ * For more discussion on check_deadline(), see: https://github.com/lf-lang/lingua-franca/issues/403
+ *
+ * @author Hokeun Kim (hokeunkim@berkeley.edu)
+ * @author Edward A. Lee (eal@berkeley.edu)
+ * @author Marten Lohstroh (marten@berkeley.edu)
+ */
+target C {
+    fast: true,
+    files: ["/lib/c/reactor-c/core/utils/vector.h",
+            "/lib/c/reactor-c/core/utils/vector.c"]
+};
+
+preamble {=
+#include "vector.c"
```

Comment:
We should include the `.h` file, not the `.c` file.

Reply:
Including the .h files requires additional steps, apparently:
1. Create a CMake file.
2. Include the CMake file.
This seems like quite a bit of extra hassle to me.  Any way to streamline this?

Reply:
I agree that it's a hassle. I'm not sure how to improve this, but @cmnrd might have ideas. I think that we should go through the hassle or make improvements to reduce the hassle, not take shortcuts and settle on a bad practice because it's quick.

Reply:
I fixed this issue by adding the cmake for vector in reactor-c, as I can imagine the resizable vector can be useful in other LF programs as well as this one.

Reply:
I think that for C and C++ requiring this additional cmake file isn't such a big hassle. One would need to do the same when writing a standalone C/C++ application.

We could think about adding syntax (or target properties) that make the inclusion of extra files simpler, but as always with C/C++ it is impossible to cover all the corner cases. I am in favor of giving programmers the tools they are used to rather than having oversimplified built-in support.